docker_push.sh

```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
-VERSION="1.2.5"
+set -e
+
+VERSION="$(git describe --tags --match 'v*')"  # Get version tag from git
+VERSION=${VERSION#v}                           # Remove leading 'v'
+
+read -p "This action will build and publish docker image version $VERSION (this will automatically proceed in 15 seconds)" -t 15
```

Comment:
If the 15 second timeout occurs, read will return a non-zero exit status that will cause the script to terminate before the push due to `set -e`. Also, this will error (and terminate the script due to `set -e`) if stdin isn't present or has been redirected from `/dev/null`. I'm not sure exactly what sort of environment our automated systems might provide, but it might be safer to just `echo` the warning, and `sleep 15`.

Reply:
Obviously `echo` + `sleep 15` doesn't allow the user to quickly bypass the pause. (But hey, maybe we want people to have time to think long and hard about the push.) But if we want the bypass capability, we could do the `read` followed by `|| true` so it is immune to the `set -e`.

Reply:
TBH, I'm not sure if I'm correct, but I guess waiting doesn't make sense in an automated setup. So, maybe provide an optional argument for the script (e.g. `-b` or `--batch`), where you just `echo` and run the command without waiting. 

And when running without that option, you might wait for 15 seconds or you might even wait forever for user's reply.

Reply:
We're not likely to use this script in automated scenarios, so it's a bit moot.